src/App.tsx

```diff
@@ -1,61 +1,126 @@
 import './App.scss';
+import { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/TODO';
+
+const idForNewTodo = Math.max(...todosFromServer.map(todo => todo.id)) + 1;
+
+export function getUserById(userId: number) {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+export const App: React.FC = () => {
+  const initialTodoState = {
+    id: idForNewTodo,
+    title: '',
+    completed: false,
+    userId: 0,
+  };
+
+  const initialTodos = [...todosFromServer];
+
+  const [newTodo, setNewTodo] = useState(initialTodoState);
+  const [todos, setTodos] = useState(initialTodos);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const handleOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTodo((prevValue) => ({ ...prevValue, title: event.target.value }));
+    setHasTitleError(false);
+  };
+
+  const handleUserChange
+= (event: React.ChangeEvent<HTMLSelectElement>) => {
+  setNewTodo((prevValue) => ({ ...prevValue, userId: +event.target.value }));
+  setHasUserIdError(false);
+};
```

Comment:
you can merge two methods into one
```suggestion
const handleInputChange = (key, value) => {
  setNewTodo(prev => ({...prev, [key]: value}))
  setHasError(prev => ({...prev, [key]: false}));
}
```